helix-term/src/ui/editor.rs

```diff
@@ -1029,9 +1029,11 @@ impl EditorView {
                 if doc
                     .selection(view.id)
                     .primary()
-                    .fragment(doc.text().slice(..))
-                    .width()
-                    <= 1
+                    .slice(doc.text().slice(..))
+                    .chars()
```

Comment:
`.slice(..).len_chars()` should be enough?

Reply:
Not really. `len_chars` doesn't take into account if the character is zero width. That's what the `width` below is for. I changed the previous implementation because it was linear with the size of the selection. I admit my change is a bit ugly though, so maybe it's worth keeping the old / slower implementation for less complexity.

Reply:
We always display zero width characters as at least of width 1 so that they are editable: https://github.com/helix-editor/helix/blob/aa4394ce1291b0af0b6da5e54cfa60a4c13f7aab/helix-core/src/graphemes.rs#L10-L32

Reply:
Ah, ok. The code previously used the `unicode_width` crate so I thought that was necessary. But it seems we can kill two birds with one stone here :)